src/core/namespace.c

```diff
@@ -287,12 +288,19 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
-                break;
+                r = lstat(m->path, &target);
+                if (r != 0)
+                        return -errno;
 
+                what = mode_to_inaccessible_node(target.st_mode);
+                if (what == NULL) {
+                        log_error("File type not supported. Note that symlinks are not allowed");
```

Comment:
so far none of the functions here logged on their own, but let the caller do that. Hence, please return a useful error here and let the caller log about it here too.

In systemd we try to relatively strictly follow the rule that functions either log all their errors or none (in which case they leave it to their caller to log). More "library-like" functions tend to let the caller log, and more "main-program-like" functions tend to log on their own. This function has so far been considered one of the former, not the latter, hence it should not log. (Note that debug logging, i.e. log_debug() instead of log_error() is always acceptable though). Also see CODING_STYLE about this.